src/CaptureClient/CaptureEventProcessor.cpp

```diff
@@ -550,12 +552,33 @@ void CaptureEventProcessorForListener::ProcessThreadNamesSnapshot(
   }
 }
 
+[[nodiscard]] static ThreadStateSliceInfo::WakeupReason toWakeupReason(
+    orbit_grpc_protos::ThreadStateSlice::WakeupReason reason) {
+  switch (reason) {
+    case orbit_grpc_protos::ThreadStateSlice::WakeupReason::
+        ThreadStateSlice_WakeupReason_kNotApplicable:
```

Comment:
That's better. Thank you for that change. But the short `orbit_grpc_protos::ThreadStateSlice::WakeupReason::kNotApplicable` should also work.

Same for the others.